### PR TITLE
tests: add tests for SetToMajorityValue block

### LINKED ISSUE #128: Add test cases for SetToMajorityValue
Add tests for:
* Replacing all values in a column with the mode
* Handling tie cases in mode (if any)
* Ensuring output dataset maintains structure

### PR DESCRIPTION
**Pull request by shivchander**
- Introduced a new test file for the SetToMajorityValue block, covering various scenarios including basic functionality, handling of numeric and mixed type columns, and edge cases such as empty columns and ties.
- Ensured that the output dataset maintains the same structure as the input, verifying that non-target columns remain unchanged.

Resolves #128

### CODE REVIEW

File: tests/blocks/utilblocks/test_settomajority.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+
+def test_set_to_majority_empty_column():
+    """Test behavior with empty column."""
+    dataset = Dataset.from_dict({"empty_col": []})
+    block = SetToMajorityValue(block_name="test_block", col_name="empty_col")
+
+    with pytest.raises(KeyError):
+        block.generate(dataset)
+
```
Comment (Maxusmusti) [inline tests/blocks/utilblocks/test_settomajority.py:66]: Does this test need any assertion to evaluate empty column?

Comment (Maxusmusti) [reply tests/blocks/utilblocks/test_settomajority.py:66]: KeyError expected, so checks out

**Review by Maxusmusti — APPROVED** [review]
